aiidalab/app.py

```diff
@@ -0,0 +1,647 @@
+# -*- coding: utf-8 -*-
+"""Module to manage AiiDA lab apps."""
+
+import re
+import os
+from os import path
+from time import sleep
+from collections import OrderedDict
+
+import requests
+import ipywidgets as ipw
+from dulwich.repo import Repo
+
+
+class AppNotInstalledException(Exception):
+    pass
+
+
+class Version(ipw.VBox):
+    """Class to choose app's version."""
+
+    def __init__(self):
+        self.change_btn = ipw.Button(description="choose seleted")
+        self.selected = ipw.Select(
+            options={},
+            description='Select version',
+            disabled=False,
+            style={'description_width': 'initial'},
+        )
+        self.info = ipw.HTML('')
+        super().__init__([self.selected, self.change_btn, self.info])
+
+
+class AiidalabApp():  # pylint: disable=attribute-defined-outside-init,too-many-public-methods,too-many-instance-attributes
```

Comment:
The term "AiiDA lab" is not a single word, so this should probably be:
```suggestion
class AiidaLabApp():  # pylint: disable=attribute-defined-outside-init,too-many-public-methods,too-many-instance-attributes
```

Reply:
One question here. It's a minor thing, but still.

Previously, we had quite a number of discussions on how to name the service, and we came up with 'AiiDA lab'. So lab should be lower case. I think the same should apply to the class names.

Reply:
We had an offline discussion and agreed that the classes should be named according to the widely adopted camel case notation.

Reply:
done